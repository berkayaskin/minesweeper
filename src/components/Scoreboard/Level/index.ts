export { Level } from './Level'
export type { LevelProps } from './Level'
