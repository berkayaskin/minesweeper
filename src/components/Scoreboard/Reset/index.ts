export { Reset } from './Reset'
export type { ResetProps } from './Reset'
