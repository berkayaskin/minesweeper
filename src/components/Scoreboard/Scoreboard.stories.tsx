import { Meta, Story } from '@storybook/react/types-6-0'
import { Scoreboard, ScoreboardProps } from './Scoreboard'

export default {
  title: 'Scoreboard/Scoreboard',
  component: Scoreboard,
} as Meta

const Template: Story<ScoreboardProps> = (args) => <Scoreboard {...args} />

export const ScoreboardExample = Template.bind({})
ScoreboardExample.args = {
  time: '000',
  levels: ['Beginner', 'Intermediate', 'Expert'],
  onReset: () => null,
  bombs: '010',
}
