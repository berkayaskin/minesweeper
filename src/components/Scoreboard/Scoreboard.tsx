import { Flex } from '@chakra-ui/react'
import styled from '@emotion/styled'
import { ChangeEvent } from 'react'
import { Counter as CounterComponent } from './Counter'
import { Level as LevelComponent } from './Level'
import { Reset as ResetComponent } from './Reset'

export interface ScoreboardProps {
  /**
   * Timer
   */
  time: string
  /**
   * Possible game difficulty levels
   * @example ['Beginner', 'Intermediate', 'Expert']
   */
  levels: string[]
  /**
   * Action handler when the GameReset button is clicked
   */
  onReset: () => void

  /**
   * Action handler when select new lvl
   */
  onChange: (event: ChangeEvent<HTMLSelectElement>) => void

  /**
   * Bombs in the field
   */
  bombs: string
}

export const Scoreboard = ({ time, levels, bombs, onReset, onChange }: ScoreboardProps) => {
  return (
    <Wrapper>
      <CounterComponent>{time}</CounterComponent>
      <Flex>
        <LevelComponent onChange={onChange}>{levels}</LevelComponent>
        <ResetComponent onReset={onReset} />
      </Flex>
      <CounterComponent>{bombs}</CounterComponent>
    </Wrapper>
  )
}

const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  padding: 0.5vw 1vw 0.25vw 1vw;
`
