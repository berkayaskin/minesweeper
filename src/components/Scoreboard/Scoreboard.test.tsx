import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { Scoreboard } from './Scoreboard'
describe('Scoreboard test cases', () => {
  it('Scoreboard renders correctly', () => {
    const { asFragment } = render(
      <Scoreboard
        time='000'
        levels={['beginner', 'intermediate', 'expert']}
        bombs='010'
        onReset={() => null}
        onChange={() => null}
      />,
    )

    expect(asFragment()).toMatchSnapshot()
  })
  it('Scoreboard select level handler check', () => {
    const onChange = jest.fn()

    render(
      <Scoreboard
        time='000'
        levels={['beginner', 'intermediate', 'expert']}
        bombs='010'
        onReset={() => null}
        onChange={onChange}
      />,
    )

    userEvent.selectOptions(screen.getByRole('combobox'), 'expert')

    expect(screen.getByRole('option', { name: 'expert' })).toBeEnabled()

    expect(onChange).toHaveBeenCalled()
  })
})
