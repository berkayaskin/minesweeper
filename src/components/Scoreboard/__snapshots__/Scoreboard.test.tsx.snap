// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scoreboard test cases Scoreboard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1gg7ynz"
  >
    <div
      class="css-d2q1pa"
    >
      000
    </div>
    <div
      class="css-k008qs"
    >
      <select
        class="css-1lpmfg0"
      >
        <option
          class="css-zcmdxr"
          value="beginner"
        >
          beginner
        </option>
        <option
          class="css-zcmdxr"
          value="intermediate"
        >
          intermediate
        </option>
        <option
          class="css-zcmdxr"
          value="expert"
        >
          expert
        </option>
      </select>
      <button
        class="css-ko2ctx"
      >
        ðŸ™‚
      </button>
    </div>
    <div
      class="css-d2q1pa"
    >
      010
    </div>
  </div>
</DocumentFragment>
`;
